dashboard/app/views/levels/_project_ui.html.haml

```diff
@@ -0,0 +1,114 @@
+- if @level.properties['free_play'] == 'true'
+  %a#create-project{href: '#'} Create project
+  %span#project-name
+  %span#save-project
+  %form#project-form{style: 'display: none'}
+    Rename:
+    %input#name{type: 'text'}
+    %input{type: 'submit'}
+- else
+  Not freeplay, cannot create project
+
+%script{src: '/shared/js/apps_api.js'}
+
+:javascript
+  var apps = storageApps();
+  var currentApp = {};
+
+  var displayCurrentApp = function() {
+    console.log(currentApp);
+    $('#create-project').hide();
+    $('#project-name').text(currentApp.name);
+    $('#project-form input#name').val(currentApp.name);
+    $('#project-form').show();
+    $('#save-project').show();
+  }
+
+  $(document).ready( function() {
+    if (window.location.hash && window.location.hash.length > 1) {
+      // try to load a project from the # param
+      apps.fetch(window.location.hash.slice(1),
+        function(data) {
+          if (data) {
+            currentApp = data;
+            console.log("fetched");
+            displayCurrentApp();
+          } else {
+            console.log('failed to fetch project ' + window.location.hash);
+          }
+        });
+    }
+  });
+
+  $(document).bind('ajaxSuccess', function(event, response) {
+    // if we have a project open, save it when receiving a milestone response from the server
+    // TODO make this an explicit event instead of just listening for all ajaxSuccess
+    if (response &&
+        response.status == 200 &&
+        response.responseJSON &&
+        response.responseJSON.level_source_id &&
+        currentApp &&
+        currentApp.id ) {
+
+      $('#save-project').text("Saving...");
+
+      currentApp.levelSourceId = response.responseJSON.level_source_id;
+      currentApp.levelId = appOptions.levelId; // TODO this is not actually a unique identifier for the level
+      currentApp.levelUrl = window.location.origin + window.location.pathname; // TODO this is hacky.. we should just have a /project url
+      currentApp.updatedAt = new Date;
+
+      apps.update(
+        currentApp.id,
+        currentApp,
+        function(success) {
+          if (success) {
+            $('#save-project').text("Saved " + currentApp.updatedAt);
+          } else {
+            $('#save-project').text("Failed to save.");
+          }
+        }
+      );
+    }
+  });
+
+   $('body').on('click', '#create-project', function() {
+    apps.create({
+      name: "Untitled",
+      createdAt: new Date,
+      updatedAt: new Date,
+      levelId = appOptions.levelId; // TODO this is not actually a unique identifier for the level
+      levelUrl = window.location.origin + window.location.pathname; // TODO this is hacky.. we should just have a /project url
```

Comment:
yeah, oops, did I push that already...

On Mon, Feb 2, 2015 at 1:20 PM, Josh Lory <notifications@github.com> wrote:

> In dashboard/app/views/levels/_project_ui.html.haml
> <https://github.com/code-dot-org/code-dot-org/pull/1185#discussion_r23959704>
> :
>
> > +            $('#save-project').text("Saved " + currentApp.updatedAt);
> > +          } else {
> > +            $('#save-project').text("Failed to save.");
> > +          }
> > +        }
> > +      );
> > +    }
> > +  });
> > +
> > +   $('body').on('click', '#create-project', function() {
> > +    apps.create({
> > +      name: "Untitled",
> > +      createdAt: new Date,
> > +      updatedAt: new Date,
> > +      levelId = appOptions.levelId; // TODO this is not actually a unique identifier for the level
> > +      levelUrl = window.location.origin + window.location.pathname; // TODO this is hacky.. we should just have a /project url
>
> I think these are meant to be hash items
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/code-dot-org/code-dot-org/pull/1185/files#r23959704>.
>